src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -248,6 +248,19 @@ internal static void FailFast(string message, Exception? exception, RhFailFastRe
                     outputMessage = message;
                 }
 
+                if (outputMessage != null)
+                {
+                    // Try to save the exception stack trace in a buffer on the stack.  If the exception is too large, we'll truncate it.
+                    const int MaxStack = 1024;
+                    Span<char> exceptionStack = stackalloc char[MaxStack];
+
+                    int length = Math.Max(MaxStack, outputMessage.Length);
+                    outputMessage.AsSpan().Slice(0, length).CopyTo(exceptionStack);
+                    // Fill the rest of the buffer with nulls
+                    if (length < MaxStack)
+                        exceptionStack.Slice(length).Clear();
```

Comment:
We should also create a record near top of the stack that allows us to quickly locate check for presence of the information. Something like:
```
struct CrashDumpRecord
{
    fixed byte Cookie[20];
    int Type;
    void* Data;
}

CrashDumpRecord* pCrashDumpRecord = (CrashDumpRecord*)stackalloc byte[sizeof(CrashDumpRecord)];
"...10 random bytes...NETRUNTIME"u8.CopyTo(...pCrashDumpRecord->Cookie...);
pCrashDumpRecord->Type = 1;
pCrashDumpRecord->Data = Unsafe.AsPointer(ref Marshal.GetReference(exceptionStack));
pCrashDumpRecord->Length = length;
```